common/src/main/java/io/honeycomb/opentelemetry/EnvironmentConfiguration.java

```diff
@@ -174,19 +186,45 @@ public static void enableOTLPTraces() {
         final String endpoint = getHoneycombTracesApiEndpoint();
         final String apiKey = getHoneycombTracesApiKey();
         final String dataset = getHoneycombTracesDataset();
+        final String serviceName = getServiceName();
 
-        if (!isPresent(apiKey)) {
-            System.out.printf("WARN: %s%n", getErrorMessage("API key", HONEYCOMB_API_KEY));
+        // helpful to know if service name is missing
+        if (!isPresent(serviceName)) {
+            System.out.printf("WARN: %s%n",
+            getErrorMessage("service name", SERVICE_NAME) +
+            " If left unset, this will show up in Honeycomb as unknown_service:java");
         }
-        if (!isPresent(dataset)) {
-            System.out.printf("WARN: %s%n", getErrorMessage("dataset", HONEYCOMB_DATASET));
+
+        // heads up: even if dataset is set, it will be ignored
+        if (isPresent(apiKey) && !isLegacyKey(apiKey) && isPresent(dataset)) {
+            if (isPresent(serviceName)) {
+                System.out.printf("WARN: Dataset is ignored in favor of service name. Data will be sent to service name: %s%n", serviceName);
+            } else {
+                // should only get here if missing service name and dataset
+                System.out.printf("WARN: Dataset is ignored in favor of service name.%n");
+            }
         }
 
         System.setProperty("otel.exporter.otlp.traces.endpoint", endpoint);
-        System.setProperty("otel.exporter.otlp.traces.headers",
-            String.format("%s=%s,%s=%s",
-                HONEYCOMB_TEAM_HEADER, apiKey,
-                HONEYCOMB_DATASET_HEADER, dataset));
+
+        // if we have an API Key, add it to the header
+        if (isPresent(apiKey)) {
+            String header = String.format("%s=%s", HONEYCOMB_TEAM_HEADER, apiKey);
+            if (isLegacyKey(apiKey)) {
+                // if the key is legacy, add dataset to the header
+                if (isPresent(dataset)) {
+                    header += String.format("%s=%s", HONEYCOMB_DATASET_HEADER, dataset);
```

Comment:
With this string append, I _think_ we need a comma at the beginning to separate the two headers.

```suggestion
                    header += String.format(",%s=%s", HONEYCOMB_DATASET_HEADER, dataset);
```